server/tserver/src/main/java/org/apache/accumulo/tserver/FileManager.java

```diff
@@ -518,8 +512,9 @@ public void setInterruptFlag(AtomicBoolean flag) {
         }
       }
 
-      for (FileSKVIterator reader : newlyReservedReaders) {
-        String filename = getReservedReadeFilename(reader);
```

Comment:
This code getting the reserved reader filename would obtain the tablet server wide lock.

Reply:
Some more comments in the code to explain the concurrency model would probably be helpful.